src/Adapter/Product/Update/ProductBasicInformationFiller.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\PrestaShop\Adapter\Tools;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductBasicInformationCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductInput\BasicInformationInput;
+use Product;
+
+class ProductBasicInformationFiller implements ProductPropertiesFillerInterface
+{
+    /**
+     * @var int
+     */
+    private $defaultLanguageId;
+
+    /**
+     * @var Tools
+     */
+    private $tools;
+
+    /**
+     * @param int $defaultLanguageId
+     * @param Tools $tools
+     */
+    public function __construct(
+        int $defaultLanguageId,
+        Tools $tools
+    ) {
+        $this->defaultLanguageId = $defaultLanguageId;
+        $this->tools = $tools;
+    }
+
+    /**
+     * @param Product $product
+     * @param BasicInformationInput $input
+     *
+     * @return array<string, mixed>
+     */
+    public function fillUpdatableProperties(Product $product, $input): array
```

Comment:
Idk if its a good approach, mainly because we cannot type the method, but there is no other way to homogenize these services and use interface + tag them :thinking:

Reply:
Other option is to simply inject them into handler one by one. And that wouldn't be a big deal, since there will always be just ~5 of them or so.

Reply:
Personally I like the way you did it. It's a repeated task with the same pattern. Following the Open closed principle, It's clearly the good choice. IMO we shouldn't care about the number of classes or services. This is a very small price to pay to keep things well organised.